runtime/compiler/runtime/JITServerAOTCache.hpp

```diff
@@ -586,7 +660,29 @@ class JITServerAOTCacheMap
    void printStats(FILE *f) const;
 
 private:
+   static std::string buildCacheFileName(const std::string &cacheDir, const std::string &cacheName);
+
    PersistentUnorderedMap<std::string, JITServerAOTCache *> _map;
+
+   // The following set is used to keep track of the caches that are in process of being loaded from file
+   // Once a cache is loaded, it is removed from the set and added to the _map
+   PersistentUnorderedSet<std::string> _cachesBeingLoaded;
+
+   // _cachesToLoadQueue is used to order the caches that need to be loaded from file.
+   // A compilation thread dequeues the first name and starts to load the cache from file.
+   // When it finishes (either successfully or not), it deletes the name from _cachesBeingLoaded.
+   PersistentList<std::string> _cachesToLoadQueue;
+
+   // _cachesExcludedFromLoading is used to keep track of the caches that we don't want to load
+   // from file, maybe because the load operation already was attempted and failed
+   // We could also populate this set from command line options
+   PersistentUnorderedSet<std::string> _cachesExcludedFromLoading;
+
+   // _cachesToSaveQueue is used to remember the caches that need to be saved to file
+   // A compilation thread dequeues the first name and tries to save the named cache to a file,
+   // possibly overwriting an existent file
+   PersistentList<std::string> _cachesToSaveQueue;
```

Comment:
Why do we store cache names here rather than pointers to cache instances?

Reply:
I wasn't sure whether caches can be unloaded and I didn't want to hold onto a pointer that could become invalid.
It would be nice to allow cache unloading in the future. If there is no client wanting a particular cache anymore, why hold onto that memory for nothing.

Reply:
Makes sense. It should be relatively easy to implement periodic purging of caches that are unused for a certain time period from memory (possibly writing the final version to the snapshot file if it adds any methods). We should open an issue for this.